docs/enhancements/1-crds-and-kubebuilder.md

```diff
@@ -0,0 +1,380 @@
+kind: MeshFederation
+metadata:
+  name: default
+spec:
+  local:
+    # ID is a unique identifier of the FDS peer, and it's used as a suffix in its Service and ServiceAccount names.
+    id: east
+    # Network name used by Istio for load balancing.
+    network: east
+    # Optional.
+    # Specifies ingress settings and properties.
+    ingress:
+      # Optional.
+      # Local ingress type specifies how to expose exported services.
+      # Currently, only two types are supported: istio and openshift-router.
+      # If "istio" is set, then the controller assumes that the Service associated with federation ingress gateway
+      # is LoadBalancer or NodePort and is directly accessible for remote peers, and then it only creates
+      # an auto-passthrough Gateway to expose exported Services.
+      # When "openshift-router" is enabled, then the controller creates also OpenShift Routes and applies EnvoyFilters
+      # to customize the SNI filter in the auto-passthrough Gateway, because the default SNI DNAT format used by Istio
+      # is not supported by OpenShift Router.
+      type: istio
+      # Optional.
+      # Can be empty if ingress.type is openshift-router.
+      gateway:
+        # Ingress gateway selector specifies to which workloads Gateway configurations will be applied.
+        selector:
+          app: federation-ingress-gateway
+        # Optional.
+        port:
+          # Optional.
+          # Port name of the ingress gateway Service.
+          # This is relevant only when the ingressType is openshift-router, but it cannot be empty.
+          name: tls-passthrough
+          # Optional.
+          # Port of the ingress gateway Service.
+          number: 15443
+    # Optional.
+    controlPlane:
+      # Optional.
+      # Control plane namespace used to create mesh-wide resources.
+      namespace: istio-system
+  remote:
+  # ID is a unique identifier of the FDS peer, and it's used as a suffix for its Service name.
+  - id: west
+    # Network name used by Istio for load balancing.
+    network: west
+    # Optional
+    ingress:
+      # Optional
+      # Addresses contain a list of IPs or a single hostname used as addresses for imported services when meshes reside in different networks.
+      # This field is not required in single network setup.
+      addresses:
+      - 192.168.1.1
+      - 192.168.2.1
+      # Optional.
+      port: 443
+      # Optional.
+      # Remote ingress type specifies how to manage client mTLS.
+      # Currently, only two types are supported: istio and openshift-router.
+      # If "openshift-router" is set the controller applies DestinationRules with SNI compatible with OpenShift Router.
+      # If "istio" is set client mTLS settings are not modified.
+      type: istio
+status:
+  conditions:
+  - meshID: west
+    status: Connected
+  - meshID: central
+    status: Disconnected
+    lastErrorMessage: "No route to host 192.168.2.1"
```

Comment:
I would stick to [typical condition properties](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) which can be understood by external tooling.

Reply:
Thank you very much for this suggestion! I didn't know that convention, so I had no idea how to approach to this problem.